### ISSUE TITLE
The Kali machines have only 1GB of RAM when started

### ISSUE THREAD

**Issue by JensMyrup** [aau-network-security/haaukins#208]
Apparently this makes the Kali machines slow down and eventually freeze to death where only solution is to create a new lab.

**Comment by kdhageman** [comment]
At the lowest level in Haaukins' Go code, `SetRAM` might (see [here](https://github.com/aau-network-security/haaukins/blob/master/virtual/vbox/vbox.go#L230)) malfunction, but since we have no unit test in place, we simply do not know yet. 

So in order to test whether this function contains a bug, I would recommend to add a unit test to `vbox_test.go`. Since the method itself might be difficult to test, you could test the `LinkedClone` function (see [here](https://github.com/aau-network-security/haaukins/blob/master/virtual/vbox/vbox.go#L230)).

@mrturkmen06 Try to think of a test case:
- What must be in place before you call the `LinkedClone` function?
  - (For example, must there be parent VM already imported?)
  - And how can you ensure that these requirements are in place in your test code?
- What parameters should you provide to the `LinkedClone` function?
  - Hint: the parameters should probably *among other parameters* contain the `SetRAM` function somewhere, since that is the piece of code we actually want to test.
- What behavior do you expect to observe from Haaukins?
  - How can you verify this behavior in your test code?
- How do you clean up after the test?

**Comment by mrturkmen06** [comment]
* Parent VM imported and created Snapshot information set
  - The requirements to be in test code can be observed by checking parameters of actual function (the function which should be tested.)
* Context, Snapshot information and VmOpts parameters provided 
* The expected behaviour tested out by retrieving information from linkedclone using `vboxmanage showvminfo <id-of-vm>` then parsed amount of ram and cpu values which should same what I set at the beginning (by providing VmOpts parameters)
* After test, the status of test function should remain same therefore all initialised VMs and other parameters removed 

According to tests, there is nothing wrong with SetRam and SetCPU functions. In next steps, unmarshalling and creating config files functions will be tested

**Comment by mrturkmen06** [comment]
- Tested NewFrontends function [here](https://github.com/aau-network-security/haaukins/blob/571f85896752ef8417701ec8d2fb55492d69969d/store/frontend_test.go#L87) > Passed
- The other tests and functions checked, there is no obvious parsing and handling error. 
-  Additionally, when an event file is created on daemon side,  it stores correct information regarding to RAM value, but does not initialise correctly, __STOP>MODIFY>RESTART__ steps will be investigated more carefully.

**Comment by mrturkmen06** [comment]
#226 is ready to review, I think, problem is solved.

**Comment by mrturkmen06** [comment]
closing issue in favor of #226  merged into 1.5.2

**Comment by tpanum** [comment]
Reopening for the sake of TODO in 1.6.2

**Comment by tpanum** [comment]
Fixed (again) in #243
